iOS-Weather-App/Controllers/WeatherViewController/WeatherViewController.swift

```diff
@@ -0,0 +1,282 @@
+//
+//  ViewController.swift
+//  iOS-Weather-App
+//
+//  Created by pavels.garklavs on 22/02/2022.
+//
+
+import UIKit
+//import MapKit
+//import CoreLocation
+import SnapKit
+
+final class WeatherViewController: UIViewController{
+
+//    private let locationManager = CLLocationManager()
+
+    private let cityNameLabel = UILabel()
+    private let temperatureLabel = UILabel()
+    private let skyConditionLabel = UILabel()
+    private let tableButton = UIButton()
+    private let detailsStackView = UIStackView()
+    private let detailStackWindTitleLabel = UILabel()
+    private let detailStackWindValueLabel = UILabel()
+    private let detailStackHumidityTitleLabel = UILabel()
+    private let detailStackHumidityValueLabel = UILabel()
+    private let detailStackVisibilityTitleLabel = UILabel()
+    private let detailStackVisibilityValueLabel = UILabel()
+//    private let network = WeatherNetwork()
+    let weatherModel = WeatherViewControllerModel()
+
+//    private var currentLocationName: String?
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+//        getCurrentLocation{ [weak self] (success) -> Void in
+//            guard let self = self else { return }
+//            if success {
+//                guard let currentLocationName = self.currentLocationName else { return }
+//                self.setupNetwork(input: currentLocationName)
+//            }
+//        }
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupBackground()
+        setupNavigationBar()
+        setupCityNameLabel()
+        setupTemperatureLabel()
+        setupDetailsStackView()
+        setupSkyConditionLabel()
+        setupTableButton()
+    }
+
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        setupBackground()
+    }
+
+}
+
+private extension WeatherViewController {
+    func setupNavigationBar() {
+        let textAttributes = [NSAttributedString.Key.foregroundColor:UIColor.white]
+        navigationController?.navigationBar.titleTextAttributes = textAttributes
+        navigationItem.title = "Current Weather"
+        let searchButton = UIBarButtonItem(image: UIImage(systemName: "magnifyingglass"), style: .plain, target: self, action: #selector(searchTapped))
+        searchButton.tintColor = .white
+        navigationItem.rightBarButtonItem = searchButton
+    }
+
+    @objc func searchTapped() {
+        weatherModel.onSearchTapped?()
+    }
+
+    func setupBackground() {
+        UIGraphicsBeginImageContext(view.frame.size)
```

Comment:
which logic do this function?

Reply:
Determines if phone is in dark mode or light mode and takes picture accordingly to it

Reply:
i mean UIGraphicsBeginImageContext

Reply:
This was really inefficient solution to making background. I fixed it and did it proper way, not converting image to bitmap